hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -360,7 +360,20 @@ int QuectelNcpClient::updateFirmware(InputStream* file, size_t size) {
 }
 
 int QuectelNcpClient::dataChannelWrite(int id, const uint8_t* data, size_t size) {
-    return muxer_.writeChannel(QUECTEL_NCP_PPP_CHANNEL, data, size);
+    int err = muxer_.writeChannel(QUECTEL_NCP_PPP_CHANNEL, data, size);
+
+    if (err) {
+        // Make sure we are going into an error state if muxer for some reason fails
+        // to write into the data channel.
+        connectionState(NcpConnectionState::DISCONNECTED);
```

Comment:
It's also probably fine to just call disable here and in all the other `dataChannelWrite()` functions.